osu.Game.Tests/Visual/Gameplay/TestSceneSkinnableSound.cs

```diff
@@ -101,5 +107,58 @@ public void TestNonLoopingStopsWithPause()
             AddAssert("sample not playing", () => !sample.Playing);
             AddAssert("sample not playing", () => !sample.Playing);
         }
+
+        [Test]
+        public void TestSkinChangeDoesntPlayOnPause()
+        {
+            DrawableSample sample = null;
+            AddStep("start sample", () =>
+            {
+                skinnableSound.Play();
+                sample = skinnableSound.ChildrenOfType<DrawableSample>().Single();
+            });
+
+            AddAssert("sample playing", () => sample.Playing);
+
+            AddStep("pause gameplay clock", () => gameplayClock.IsPaused.Value = true);
+            AddUntilStep("wait for sample to stop playing", () => !sample.Playing);
+
+            AddStep("trigger skin change", () =>
+            {
+                skinSource.TriggerSourceChanged();
+            });
+
+            AddStep("retrieve new sample", () =>
+            {
+                DrawableSample newSample = skinnableSound.ChildrenOfType<DrawableSample>().Single();
+                Assert.IsTrue(newSample != sample, "Sample still hasn't been updated after a skin change event");
+                sample = newSample;
+            });
+
+            AddAssert("new sample paused", () => !sample.Playing);
```

Comment:
i don't think "pausing" samples is ever a thing.

Reply:
Urgh, will try to keep an eye out next time.